src/components/InRequest/InRequest.tsx

```diff
@@ -76,286 +76,374 @@ export const InRequest: FunctionComponent<any> = (props) => {
-    });
-  };
-
-  const onPrevOrgChange = (
-    event: ChangeEvent<HTMLInputElement>,
-    data?: InputOnChangeData
-  ) => {
-    const prevOrgVal = data?.value ? data.value : "";
-    setFormData((f: INewInForm) => {
-      return { ...f, prevOrg: prevOrgVal };
-    });
-  };
+  }, [empType]);
 
   useEffect(() => {
     // Only set the supGovLead if this is a New request
     if (props.view === INFORMVIEWS.NEW) {
       let persona: SPPersona = {};
       persona = { ...userContext.user };
-
-      setFormData((f: INewInForm) => {
-        return { ...f, supGovLead: persona };
-      });
+      // TODO - Set this to use setValue instead, and move it from being a useEffect
+      //  to just being after the if statement for determing current user
+      //  Set value only causes rerender under specific conditions, where resetField re-rerenders
+      //  more frequently.  setValue isn't working for SPPersona due to TS issues in v7 of RHF
+      resetField("supGovLead", { defaultValue: persona });
     }
-  }, [userContext.user, props.view]);
+  }, [userContext.user, props.view, resetField]);
 
-  /* If the parent component updates prop to indicate a save or cancel event, then peform appropriate actions,
-      and call the passed in parents function.  If we are successfully saving then pass the formData back to the
-      parent's callback function, otherwise, don't pass it back so that it leaves the panel open for action */
-  useEffect(() => {
-    if (props.saveCancelEvent === "save") {
-      /* TODO - Evaluate if the user has valid data to update the record with, and update the record.  If it isn't valid
-        then call the parent's callback passing no return value so that it leaves panel open */
-      if (true /*Is Data Valid? */) {
-        props.onEditSaveCancel(formData);
-      } else {
-        /* Data not valid, so tell parent not to close panel  */
-        //props.onEditSaveCancel();
-      }
-    } else if (props.saveCancelEvent === "cancel") {
-      props.onEditSaveCancel();
-    }
-  }, [props.saveCancelEvent, formData, props]);
-
-  /* The data based on the ReqId prop that is passed in */
+  /* Get the data based on the ReqId prop that is passed in */
   useEffect(() => {
     const loadRequest = async () => {
       const res = await requestApi.getItemById(props.ReqId);
       if (res) {
         setFormData(res);
+        // Populate the React-Hook-Form with the data
+        reset(res);
       }
     };
 
     loadRequest();
-  }, [props.ReqId, requestApi]);
+  }, [props.ReqId, requestApi, reset]);
 
+  /* If they change employee type, clear out the selected grade */
   useEffect(() => {
-    updateGradeRankOpt(formData?.empType as emptype);
-  }, [formData.empType]);
+    setValue("gradeRank", "");
+  }, [empType, setValue]);
 
   /* Temporarily show a Loading screen if we don't have the current user info yet. */
   if (userContext.loadingUser) {
     return <>Loading...</>;
+  } else {
   }
```

Comment:
I'll take a look at moving some of the functions down below it.  Unfortunately, we saw the other day that the useEffects have to come before it, but it might be that if I have all useEffects after it, then it will be okay, as the order/number of the useEffects will always be the same and not vary based on this conditional.

Reply:
Moved some functions up.  Also see potential issue with how userContext is working.  Updated to no longer key of loadingUser which may return a value prior to the value of the user being set.  Instead key off userContext.user having a value.